src/logic/game_settings.cc

```diff
@@ -36,7 +36,7 @@ Widelands::PlayerNumber GameSettings::find_shared(const PlayerSlot slot) const {
 		return result;
 	}
 	result = 1;
-	for (; result <= players.size(); ++result) {
+	for (; result <= static_cast<Widelands::PlayerNumber>(players.size()); ++result) {
```

Comment:
well I believe thgis cast is misleading as the size of the players table is definitly not a PlayerNumber

Reply:
@hessenfarmer I used that cast because on L33, 'result' is declared as 'Widelands::PlayerNumber'; the codeql error was something about comparison between different types. So I took my best guess there. I figured 'result' should never be greater than the max of what the Widelands::PlayerNumber type would allow.

But I'm open to suggestions if there's a better fix.

Reply:
well I think casting it to an integer type of similar size as Player number would be a solution without decreasing readability